File: BestPrice/src/main/java/pe/edu/upc/bestprice/entities/Tienda.java
package pe.edu.upc.bestprice.entities;

import jakarta.persistence.*;

import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name="Tienda")
public class Tienda {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idTienda;
    @Column(name="nombreTienda", nullable = false, length = 50)
    private String nombreTienda;

    public Tienda() {
    }

    public Tienda(int idTienda, String nombreTienda, String detalleTienda, String numeroTelefono, boolean estadoTienda, LocalDate fechaTienda, TipoTienda tipoTienda, List<Producto> productosTienda) {
        this.idTienda = idTienda;
        this.nombreTienda = nombreTienda;
    }

    public int getIdTienda() {
        return idTienda;
    }

    public void setIdTienda(int idTienda) {
        this.idTienda = idTienda;
    }

    public String getNombreTienda() {
        return nombreTienda;
    }

    public void setNombreTienda(String nombreTienda) {
        this.nombreTienda = nombreTienda;
    }
}

File: BestPrice_Front/BestPrice/src/app/models/tienda.ts
import { LineatiendaInsert } from "./lineatiendainsert"
import { TipoTienda } from "./tipotienda"

export class Tienda {
    idTienda: number = 0
    nombreTienda: string = ""
    descripcionTienda: string = ""
    numeroTelefono: string = ""
    estadoTienda: boolean = false
    createdAtLineaTienda: Date=new Date()
    lineaTienda: LineatiendaInsert = new LineatiendaInsert()
    tipoTienda: TipoTienda = new TipoTienda()
}

